mozetl/health_check/normandy/job.py

```diff
@@ -0,0 +1,204 @@
+    recipe_health = drop_normandy_prefix(
+        crosstab_to_dict(
+            events.select(events.event_map_values["source"], events.event_string_value)
+            .filter("event_map_values['source'] LIKE 'normandy/recipe/%'")
+            # FIXME: https://bugzilla.mozilla.org/show_bug.cgi?id=1543817
+            # We can maybe drop this after 68 (ESR) rolls out
+            .replace("custom_2_error", "backoff")
+            .crosstab("event_map_values[source]", "event_string_value")
+        )
+    )
+    action_health = drop_normandy_prefix(
+        crosstab_to_dict(
+            events.select(events.event_map_values["source"], events.event_string_value)
+            .filter("event_map_values['source'] LIKE 'normandy/action/%'")
+            .crosstab("event_map_values[source]", "event_string_value")
+        )
+    )
+    runner_nested_dict = crosstab_to_dict(
+        events.select(events.event_map_values["source"], events.event_string_value)
+        .filter("event_map_values['source'] = 'normandy/runner'")
+        .crosstab("event_map_values[source]", "event_string_value")
+    )
+    assert len(runner_nested_dict) == 1
+    # only one runner, so discard "outer" dict
+    runner_health = runner_nested_dict["normandy/runner"]
+    status_totals = dict(
+        [
+            (row[0], row[1])
+            for row in events.groupBy(events.event_string_value)
+            .agg(F.count("*"))
+            .collect()
+        ]
+    )
+
+    def myzip(submission_date, value):
+        return zip(submission_date, value)
+
+    zipped_schema = T.ArrayType(
+        T.StructType(
+            [
+                T.StructField("submission_date", T.StringType()),
+                T.StructField("value", T.LongType()),
+            ]
+        )
+    )
+    myzip = F.udf(myzip, zipped_schema)
+
+    flat_scalar = (
+        longitudinal.select(
+            "submission_date", F.explode("scalar_parent_normandy_recipe_freshness")
+        )
+        .select("key", myzip("submission_date", "value").alias("time_value"))
+        .select("key", F.explode("time_value").alias("time_value"))
+    )
+    flat_scalar = flat_scalar.filter(
+        "time_value.submission_date >= '%s'" % start_longitudinal.strftime("%Y-%m-%d")
+    ).filter("time_value.value IS NOT NULL")
+    recipe_versions = nest_keys(
+        dict(
+            [
+                ((row[0], row[1]), row[2])
+                for row in flat_scalar.groupBy("key", "time_value.value")
+                .count()
+                .collect()
+            ]
+        )
+    )
+    return NormandyHealthCheckResult(
+        now, recipe_versions, recipe_health, action_health, runner_health, status_totals
+    )
+
+
+@click.command()
+@click.option("--period", default=6, help="length of the retention period in hours")
+@click.option(
+    "--slack", default=6, help="number of hours to account for submission latency"
+)
+def main(period, slack):
```

Comment:
I would consider putting this function outside of click, so it's easier to run the code from e.g. a databricks notebook. You can see an example of this in search aggregates:

https://github.com/mozilla/python_mozetl/blob/ad257f64c5de876b615e7ba56608d0c91d7d2fe6/mozetl/search/aggregates.py#L390